### PR TITLE
Min length validator

### PR DESCRIPTION
**Pull request by stuarthicks**
A new validator to specify a constraint on the minimum length of a string.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -5,45 +5,51 @@ endif
 
 .DEFAULT_GOAL = all
 
+ifdef VERBOSE
+V = -v
+else
+.SILENT:
```
Comment (mbowBC) [inline Makefile]: 👍

File: gkgen/minlength_test.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+		s := minLengthStruct{String: "foo"}
+		st := reflect.TypeOf(s)
+		field, found := st.FieldByName("String")
+		assert.True(found)
+
+		_, err := validator.Generate(field, []string{})
+		assert.Error(err)
+	})
+
+	t.Run("RequireParamIsANumber", func(t *testing.T) {
```
Comment (mbowBC) [inline gkgen/minlength_test.go]: can you clarify naming of this test.

Comment (mbowBC) [inline gkgen/minlength_test.go]: ok nw i think i get it now testing []string{"wat"} and not minLengthStruct{int: 3}

File: gokay/length_test.go [hunk 3]
```diff
@@ -29,3 +29,38 @@ func TestLengthSlice(t *testing.T) {
...
+
+func TestMinLengthString(t *testing.T) {
+	var validCases = []struct {
+		String    string
+		MinLength int64
+	}{
+		{String: "foo", MinLength: 3},
+		{String: "", MinLength: 0},
+		{String: "a", MinLength: 1},
+		{String: "a", MinLength: 100},
```
Comment (mbowBC) [inline gokay/length_test.go]: does this not fail on ```if expected < int64(len(*str)) {```

Comment (mbowBC) [inline gokay/length_test.go]: Length was 1, needs to be at least 100, len(*str), expected
I think it's picking up the address of the pointer to check

File: gokay/length.go [hunk 4]
```diff
@@ -18,3 +18,15 @@ func LengthString(expected int64, str *string) error {
...
 	return nil
 }
+
+// MinLengthString checks if the value of a string pointer has a length of at least 'expected'
+func MinLengthString(expected int64, str *string) error {
+	if str == nil {
+		return nil // Covers the case where a value can be Nil OR has a length constraint
+	}
+
+	if expected < int64(len(*str)) {
```
Comment (mbowBC) [inline gokay/length.go]: change to >

Comment (stuarthicks) [inline gokay/length.go]: Whoops 😛

**Comment by mbowBC** [comment]
LGTM now, you can merge it when you've done your fab-common change.

**Pull request merged**